src/components/SpecialResolution/SpecialResolutionForm.vue

```diff
@@ -0,0 +1,463 @@
+  }
+  /** Called to add a new date. */
+  async onResolutionDateSync (val: string): Promise<void> {
+    await this.$nextTick()
+    this.resolutionDateText = val
+    this.setResolution({
+      ...this.getCreateResolutionStep,
+      resolutionDate: val
+    })
+  }
+  /** called to add new resolutionDateText. */
+  protected onResolutionTextChanged (val: string) {
+    this.setResolution({
+      ...this.getCreateResolutionStep,
+      resolutionText: val
+    })
+  }
+
+  /** Validation rule for name */
+  nameRules (label, isRequired = true): Array<Function> {
+    return [
+      v => isRequired ? (!!v?.trim() || `${label} is required`) : true,
+      v => (v?.length <= 30) || 'Cannot exceed 30 characters' // maximum character count
+    ]
+  }
+
+  /** Validation rule for individual name fields */
+  firstNameRules = this.nameRules('First Name');
+  middleNameRules = this.nameRules('Middle Name', false);
+  lastNameRules = this.nameRules('Last Name');
+
+  /** The minimum date that can be entered (resolution date). */
+  get signatureDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.signatureDateMin)
+  }
+
+  /** The minimum date that can be entered (resolution date). */
+  get signatureDateMin (): Date {
+    if (this.resolutionDateText) {
+      const resolutionDate = this.yyyyMmDdToDate(this.resolutionDateText)
+      return resolutionDate
+    }
+    const date = new Date()
+    date.setHours(0, 0, 0, 0)
+    return date
+  }
+
+  /** The maximum date that can be entered (today). */
+  get signatureDateMaxStr (): string {
+    const date = this.signatureDateMax
+    const dateStr = this.dateToYyyyMmDd(date)
+    return dateStr
+  }
+
+  /** The maximum date that can be entered (today). */
+  private get signatureDateMax (): Date {
+    const date = new Date()
+    return date
+  }
+
+  /** Validations rules for signing date field. */
+  get signatureDateRules (): Array<Function> {
+    return [
+      (v: string) => !!v || 'Signature date is required',
+      (v: string) =>
+        this.isBetweenDates(this.signatureDateMinStr,
+          this.signatureDateMaxStr,
+          v) ||
+        `Date should be between ${this.dateToPacificDate(this.signatureDateMin, true)} and
+         ${this.dateToPacificDate(this.signatureDateMax, true)}`
+    ]
+  }
+  /** called to add new signature date  */
+  async onSigningDateSync (val: string): Promise<void> {
+    await this.$nextTick()
+    this.signatureDateText = val
+    this.setResolution({
+      ...this.getCreateResolutionStep,
+      signingDate: val
+    })
+  }
+  /** called to store sining party. */
+   @Watch('signingPerson', { deep: true })
+  protected async onSigningPersonChanged (): Promise<void> {
+    await this.$nextTick()
+    this.setResolution({
+      ...this.getCreateResolutionStep,
+      signingPerson: this.signingPerson
+    })
+  }
+   /** Set values if exist */
+   created () {
```

Comment:
Created should be protected, and our code style is to place it near the top of the class (possibly just under the local getters, which are kind of like variables).

Reply:
Now that I look at what this does, can't you do it during the variable initializations?

If you depend on the getter, note that these will be assigned the getter values as of when this component is created. They will not be assigned a subsequent getter value (not reactive).